File: src/main/java/yunas/db/factory/ConnectionPoolFactory.java
package yunas.db.factory;

import yunas.configuration.Configuration;

import javax.sql.DataSource;
import java.util.concurrent.ConcurrentHashMap;

/**
 * Create ConnectionPool with HicariCP.
 *
 */
public interface ConnectionPoolFactory {

    public ConcurrentHashMap<String, DataSource> create(Configuration conf);

}

File: src/main/kotlin/yunas/db/factory/ConnectionPoolFactory.kt
package yunas.db.factory

import yunas.configuration.Configuration

import javax.sql.DataSource
import java.util.concurrent.ConcurrentHashMap

/**
 * Create ConnectionPool with HicariCP.
 *
 */
interface ConnectionPoolFactory {

    fun create(conf: Configuration): ConcurrentHashMap<String, DataSource>

}
